pvlib/bifacial/infinite_sheds.py

```diff
@@ -139,19 +138,24 @@ def _vf_row_sky_integ(f_x, surface_tilt, gcr, npoints=100):
     """
     # handle Series inputs
     surface_tilt = np.array(surface_tilt)
-    cst = cosd(surface_tilt)
     # shaded portion
     x = np.linspace(0, f_x, num=npoints)
     psi_t_shaded = masking_angle(surface_tilt, gcr, x)
-    y = 0.5 * (cosd(psi_t_shaded) + cst)
+    y = 0.5 * (cosd(psi_t_shaded + surface_tilt) + 1)
     # integrate view factors from each point in the discretization. This is an
     # improvement over the algorithm described in [2]
-    vf_shade_sky_integ = np.trapz(y, x, axis=0)
+    vf_shade_sky_integ = np.where(
+        f_x<1e-6,
+        0.5 * (cosd(masking_angle(surface_tilt, gcr, 0.) + surface_tilt) + 1),
```

Comment:
I think this might be `y[0]` because `x` uses `np.linspace()` starting at zero:
```suggestion
        y[0],
```

Reply:
I agree, that's my explicit coding to make it easier to verify I transferred my notes to code correctly :)

These lines are going away and will be replaced by a call to the new utils function.